src/warc2zim/converter.py

```diff
@@ -648,43 +638,57 @@ def retrieve_illustration(self):
 
         Uses fallback in case of errors/missing"""
 
-        if self.favicon_url or self.favicon_path:
-            # look into WARC records
-            for record in iter_warc_records(self.inputs):
-                if record.rec_type != "response":
-                    continue
-                url = get_record_url(record)
-                path = normalize(HttpUrl(url))
-                if path == self.favicon_path or url == self.favicon_url:
-                    logger.debug("Found WARC record for favicon")
+        for favicon_url in self.favicon_urls:
+            favicon_path = normalize(favicon_url)
+            if favicon_path in self.expected_zim_items:
+                # look into WARC records to search for icon as illustration
+                for record in iter_warc_records(self.inputs):
+                    if record.rec_type != "response":
+                        continue
+                    url = get_record_url(record)
+                    path = normalize(HttpUrl(url))
+                    if path != favicon_path:
+                        continue
                     if (
                         record.http_headers
                         and record.http_headers.get_statuscode() != "200"
                     ):  # pragma: no cover
-                        logger.warning("WARC record for favicon is unusable")
+                        logger.warning(
+                            f"WARC record for favicon {favicon_path.value} is unusable"
```

Comment:
Would be the case for redirects right?

Reply:
This is indeed a limitation of current code. Never seen in the wild, so probably not something to take care of for now. This code is more targeted at 404.